test/parallel/test-stream-readable-manualStart.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+const { Readable } = require('stream');
+
+{
+  let didRead = true;
+  const readable = new Readable({
+    manualStart: false,
+    read: common.mustCall(() => {
+      didRead = true;
+      assert.strictEqual(readable.readableDidRead, true);
+    })
+  });
+
+  setImmediate(() => {
+    assert.strictEqual(didRead, true);
+    readable.on('data', common.mustNotCall);
+  });
+}
+
+{
+  let didStart = false;
+  let didRead = true;
+  const readable = new Readable({
+    manualStart: true,
+    read: common.mustCall(() => {
+      didRead = true;
+      assert.strictEqual(didStart, true);
+      assert.strictEqual(readable.readableDidRead, true);
+    })
+  });
+
+  setTimeout(() => {
+    assert.strictEqual(didRead, true);
+    didStart = true;
+    readable.on('data', common.mustNotCall);
+  }, 1);
+}
```

Comment:
Perhaps we should have a 3rd case which tests the default behavior (when this `manualStart` is not set)?